src/subdomains/staking/application/services/liquidity-management.service.ts

```diff
@@ -76,40 +82,36 @@ export class LiquidityManagementService {
   }
 
   // --- MASTERNODES ---- //
-  private async createMasternodes(count: number): Promise<void> {
+  private async checkMasternodesInProcess(): Promise<void> {
+    const allInProcessMasternodes = await this.masternodeService.getAllWithStates([
+      MasternodeState.ENABLING,
+      MasternodeState.RESIGNING,
+    ]);
+    await this.handleMasternodesWithState(allInProcessMasternodes, MasternodeState.ENABLING);
+    await this.handleMasternodesWithState(allInProcessMasternodes, MasternodeState.RESIGNING);
+
+    // TODO (Krysh) check blockchain if pre_enabled are created or pre_resigned are resigned
+  }
+
+  private async startMasternodeEnabling(count: number): Promise<void | void[]> {
     // get n addresses from the masternode table, where masternode state is idle
     const idleMasternodes = await this.masternodeService.getIdleMasternodes(count);
-
-    let tx: string;
-    for (const node of idleMasternodes) {
-      tx = await this.client.sendUtxo(
-        Config.staking.liquidityWalletAddress,
-        node.owner,
-        Config.masternode.collateral + Config.masternode.fee + Config.masternode.creationFee,
-      );
-
-      console.info(`Sending collateral to masternode wallet: ${tx}`);
-
-      if (tx) await this.client.waitForTx(tx).catch((e) => console.error(`Wait for creation TX failed: ${e}`));
-
-      await this.masternodeService.designateCreating(node.id);
-    }
+    return this.handleMasternodesWithState(idleMasternodes, MasternodeState.IDLE);
   }
 
-  private async resignMasternodes(count: number): Promise<void> {
+  private async startResignMasternodes(count: number): Promise<void | void[]> {
     const masternodes = await this.masternodeService.getOrderedByTms();
+    return this.handleMasternodesWithState(masternodes.splice(0, count), MasternodeState.ENABLED);
+  }
 
-    for (const masternode of masternodes.slice(0, count)) {
-      console.info(`Resigning masternode ${masternode.id}`);
+  private async handleMasternodesWithState(masternodes: Masternode[], state: MasternodeState): Promise<void | void[]> {
+    const filteredMasternodes = masternodes.filter((mn) => mn.state === state);
 
-      const message = Util.template(Config.masternode.resignMessage, {
-        id: masternode.id.toString(),
-        hash: masternode.creationHash,
-      });
-      const signature = await this.client.signMessage(Config.staking.liquidityWalletAddress, message);
+    const [txFunction, updateFunction] = this.receiveTxFunctionFor(state);
 
-      await this.masternodeService.prepareResign(masternode.id, signature);
-    }
+    return Promise.all(
+      filteredMasternodes.map((node) => txFunction(node).then((txId: string) => updateFunction(node, txId))),
+    ).catch(console.error);
```

Comment:
I think the catch should be inside the Promise.all, so not all parallel executions are aborted.